router-tests/testenv/testenv.go

```diff
@@ -364,22 +364,36 @@ func createTestEnv(t testing.TB, cfg *Config) (*Environment, error) {
 	}
 
 	if cfg.EnableKafka {
-		kafkaStarted.Add(1)
-		go func() {
-			defer kafkaStarted.Done()
+		if os.Getenv("CI") == "true" {
```

Comment:
This makes it impossible or hard to test locally. Why do you think running kafka in CI won't result in flaky tests? Is it all about how we instantiate Kafka?

Reply:
Kafka still runs locally, it just doesn't start a container if it's in CI. It's something to do with load from what we could tell during debugging, not due to container startup or cleanup but just connection timeouts. 

I believe containers started by `services` are a bit different in GHA than ones started by docker within the job, not sure exactly how but the tests run faster and fail less when it's a service.

Reply:
FWIW the GHA service container is run with the same environment as the one it uses locally

Reply:
In summary it improves but doesn't eliminate the cause?